openfl-tutorials/interactive_api/HPU/PyTorch_MedMNIST_2D/workspace/HPU_Pytorch_MedMNIST_2D.ipynb

```diff
@@ -0,0 +1,627 @@
+    "        for inputs, targets in val_loader:\n",
+    "            outputs = model(inputs.to(device))\n",
+    "            \n",
+    "            if task == 'multi-label, binary-class':\n",
+    "                targets = targets.to(torch.float32).to(device)\n",
+    "                loss = criterion(outputs, targets)\n",
+    "                m = nn.Sigmoid()\n",
+    "                outputs = m(outputs).to(device)\n",
+    "            else:\n",
+    "                targets = torch.squeeze(targets, 1).long().to(device)\n",
+    "                loss = criterion(outputs, targets) \n",
+    "                m = nn.Softmax(dim=1)\n",
+    "                outputs = m(outputs).to(device)\n",
+    "                targets = targets.to(torch.float32)\n",
+    "#                 targets = targets.float().resize_(len(targets), 1)\n",
+    "#                 right way to use float() is tensor.to(torch.float32) else gives RuntimeError: expected scalar type Long but found Float\n",
+    "#                 The resize_(len(targets), 1) operation in above line causes below graph compile error, hence removed it\n",
+    "#                 RuntimeError: FATAL ERROR :: MODULE:BRIDGE syn compile encountered : Graph compile failed\n",
+    "\n",
+    "            total_loss.append(loss.item())\n",
+    "            total_samples += targets.shape[0]\n",
+    "            \n",
+    "            # API call to trigger execution\n",
+    "            htcore.mark_step()\n",
+    "            \n",
+    "            pred = outputs.argmax(dim=1)\n",
+    "            val_score += pred.eq(targets).sum().cpu().numpy()\n",
+    "        \n",
+    "        acc = val_score / total_samples        \n",
+    "        test_loss = sum(total_loss) / len(total_loss)\n",
+    "\n",
+    "        return {'acc': acc, \n",
+    "                'test_loss': test_loss,\n",
+    "                }\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f0ebf2d",
+   "metadata": {},
+   "source": [
+    "## Time to start a federated learning experiment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d41b7896",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create an experimnet in federation\n",
+    "experiment_name = 'medmnist_exp'\n",
+    "fl_experiment = FLExperiment(federation=federation, experiment_name=experiment_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "41b44de9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The following command zips the workspace and python requirements to be transfered to collaborator nodes\n",
+    "fl_experiment.start(model_provider=MI, \n",
+    "                    task_keeper=TI,\n",
+    "                    data_loader=fed_dataset,\n",
+    "                    rounds_to_train=3,\n",
+    "                    opt_treatment='RESET',\n",
+    "                    device_assignment_policy='CUDA_PREFERRED')"
```

Comment:
The `envoy_config` does not contain any GPU devices registered, therefore this line does not make any difference and may be removed